### PR TITLE
Issue #172: Migration of PaymentController to OpenApi yaml schema

### PR DESCRIPTION
**Pull request by edwin9870**
## 🚀 Description

Added OpenAPI spec for PaymentController

## 📄 Motivation and Context

Fixes Issue #172 

## 🧪 How Has This Been Tested?

- Test cases are passing
- Endpoints are working

## 📦 Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api/src/main/resources/api.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ info:
   title: Swagger MyHome - OpenAPI 3.0
 tags:
   - name: Users
+  - name: Payments
+  - name: Members
+  - name: communities
```
Comment (mslowiak) [inline api/src/main/resources/api.yaml]: `communities` -> `Communities`

File: api/src/main/resources/api.yaml [hunk 2]
```diff
@@ -138,4 +248,82 @@ components:
...
+          type: string
+    ListMemberPaymentsResponse:
+      type: object
+      properties:
+        payments:
+          type: array
+          items:
+            $ref: '#/components/schemas/MemberPayment'
+          uniqueItems: true
+    AdminPayment:
```
Comment (mslowiak) [inline api/src/main/resources/api.yaml:963]: Maybe it is a good idea here to merge `AdminPayment` with `MemberPayment`, @jmprathab ?

File: service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java [hunk 3]
```diff
@@ -125,7 +126,7 @@ default EnrichedSchedulePaymentRequest enrichSchedulePaymentRequest(
         .collect(Collectors.toSet());
     return new EnrichedSchedulePaymentRequest(request.getType(),
         request.getDescription(),
-        request.isRecurring(),
+        request.getRecurring(),
```
Comment (mslowiak) [inline service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java]: I think we can add `"booleanGetterPrefix" : "is"` to the `additionalProperties` in `api/build.gradle` and the getters will be with `is` prefix without the changes :)

**Review by mslowiak — CHANGES_REQUESTED** [review]
Would be great if you get some time to read our contribution guide, because the name of pull request and commit names are incorrect with our convention.
https://github.com/jmprathab/MyHome/blob/master/CONTRIBUTING.md

File: service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java (hunk 3 above)
Comment (edwin9870) [reply service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java]: Yes, I think that would be better. May I do that change @mslowiak ?

Comment (mslowiak) [reply service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java]: @edwin9870 Yes, please :)

Comment (edwin9870) [reply service/src/main/java/com/myhome/controllers/mapper/SchedulePaymentApiMapper.java]: Done aeb99416b8d5a9f187026429376f1cb4b79ea26b

File: api/src/main/resources/api.yaml (hunk 2 above)
Comment (jmprathab) [reply api/src/main/resources/api.yaml:963]: > Maybe it is a good idea here to merge `AdminPayment` with `MemberPayment`, @jmprathab ?

Yes. That sounds good!

Comment (edwin9870) [reply api/src/main/resources/api.yaml:963]: @mslowiak, are you waiting for any action of me to merge this PR?

Comment (mslowiak) [reply api/src/main/resources/api.yaml:963]: I guess we will create another issue for that change, looks good for me so just please resolve conflicts so we can merge a PR

**Review by mslowiak — APPROVED** [review]

**Comment by mslowiak** [comment]
@edwin9870 Could you resolve conflicts please?

**Pull request merged**